### ISSUE TITLE
`description` under `parameter` is removed after creating the wftmpl or update. Linting the wftmpl results in error too.

### ISSUE THREAD

**Issue by edwardfung123** [argoproj/argo-workflows#8233]
## Checklist

 

* [x] Double-checked my configuration.
* [x] Tested using the latest version.
* [ ] Used the Emissary executor.

## Summary

What happened/what you expected to happen?

According to [the doc](https://argoproj.github.io/argo-workflows/fields/#parameter), we should be able to add `description` to a `parameter`.
After adding some human readable description to parameters of my worklow. But it is not saved. The `argo lint` also complain about the `description` field.

What version are you running? 

`3.2.9`

## Diagnostics

Paste the smallest workflow that reproduces the bug. We must be able to run the workflow.

```yaml
# copy from the example
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: hello-world-parameters-
spec:
  # invoke the whalesay template with
  # "hello world" as the argument
  # to the message parameter
  entrypoint: whalesay
  arguments:
    parameters:
    - name: message
      value: hello world
      description: "Message to be printed"
      # added this description.
  templates:
  - name: whalesay
    inputs:
      parameters:
      - name: message       # parameter declaration
    container:
      # run cowsay with that message input parameter as args
      image: docker/whalesay
      command: [cowsay]
      args: ["{{inputs.parameters.message}}"]
```

```bash
# log from the linter
argo lint desc_wftmpl.yaml 
desc_wftmpl.yaml:
   ✖ in "hello-world-parameters-" (Workflow): json: unknown field "description"

✖ 1 linting errors found!
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by alexec** [comment]
feat: support workflow template parameter description (#7309) (#7346)

**Comment by alexec** [comment]
This feature is only available in v3.3

**Comment by sarabala1979** [comment]
Can you check your CLI version?
